crypt.go

```diff
@@ -0,0 +1,105 @@
+package main
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"crypto/sha1"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"io"
+
+	"golang.org/x/crypto/pbkdf2"
+)
+
+const saltSize = 16 // bytes
+
+// Openssl's way of derive key from passphrase is considered too weak, see:
+// https://security.stackexchange.com/questions/29106/openssl-recover-key-and-iv-by-passphrase
+// Use pbkdf2 instead see:
+// https://github.com/riverrun/comeonin/wiki/Choosing-the-password-hashing-algorithm
+func deriveKey(passphrase string, salt []byte) *[32]byte {
+	dk := pbkdf2.Key([]byte("some password"), salt, 4096, 32, sha1.New)
+	var key [32]byte
+	copy(key[:], dk)
+	return &key
+}
```

Comment:
```suggestion
const keySize = 32
func deriveKey(passphrase string, salt []byte) []byte {
	return pbkdf2.Key([]byte("some password"), salt, 4096, keySize, sha1.New)
}
```

Reply:
I think this is by design, by using an array instead of a slice, we can be sure at compile time that the caller is passing in a correct length key.
See: https://github.com/gtank/cryptopasta/issues/3

Reply:
I see. Then maybe this will be better?
```
	var key [32]byte
	dk := pbkdf2.Key([]byte(passphrase), salt, 4096, len(key), sha1.New)
```

Reply:
Agreed